Make pytorch_qnnpack a shared library

smeenai opened the pull request:
Summary:
This library contains global state, e.g. pytorch_qnnp_params. If we make
it a static library, different shared libraries linking that static
library can end up with their own copies of the global state, leading to
bugs. Make it a shared library instead, to avoid this issue.

Test Plan: buck2 test fbsource//fbandroid/javatests/com/facebook/playground/apps/fb4aplayground/scenarios/pytorchscenario:pytorchscenario -- --run-disabled --regex runBundledInputWithLocalAsset

Differential Revision: D51926024




cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

malfet commented:
Caffe2 title is a bit misleading here. This PR purely affects internal builds

malfet reviewed (approved):
OSS part looks "ok" to me (please don't use [caffe2] for PyTorch changes)

malfet commented:
@smeenai just remove `[caffe2]` from the title if you don't mind

smeenai committed:
Make pytorch_qnnpack a shared library (#115570)

Summary:

This library contains global state, e.g. pytorch_qnnp_params. If we make
it a static library, different shared libraries linking that static
library can end up with their own copies of the global state, leading to
bugs. Make it a shared library instead, to avoid this issue.

Test Plan: buck2 test fbsource//fbandroid/javatests/com/facebook/playground/apps/fb4aplayground/scenarios/pytorchscenario:pytorchscenario -- --run-disabled --regex runBundledInputWithLocalAsset

Reviewed By: malfet

Differential Revision: D51926024
